src/resmom/linux/alps.c

```diff
@@ -3013,48 +3437,364 @@ parse_char_data(void *ud, const XML_Char *s, int len)
 {
 	ud_t *d;
 
-	if (!ud)
-		return;
-	d = (ud_t *)ud;
-	handler[d->stack[d->depth]].char_data(d, s, len);
+	if (!ud)
+		return;
+	d = (ud_t *)ud;
+	handler[d->stack[d->depth]].char_data(d, s, len);
+	return;
+}
+
+/**
+ * @brief
+ *	This function walks all the segments and fills in the information needed to
+ * generate the vnodes. It is called directly when PBS is using BASIL 1.2 or
+ * prior XML inventories. It is called from within the socket loop when PBS
+ * gets BASIL 1.3 and BASIL 1.4 XML inventory.
+ * @param[in] node	information about the node
+ * @param[in] nv	vnode list information
+ * @param[in] arch	the architecture of the vnode
+ * @param[in/out] total_seg	latest count of the segments, to increase
+ *				across sockets
+ * @param[in] order	order number of the vnode
+ * @param[in/out] name_buf	upon exiting it contains the name of the vnode
+ * @param[in/out] total_cpu	keeps a running count of cpus for the whole node
+ * @param[in/out] total_mem	keeps a running count of mem for the whole node
+ *
+ * These last three parameters are used for the vnode_per_numa_node = 0 case.
+ * Where we only create one PBS vnode per Cray compute node
+ *
+ * @return Void
+ */
+void
+inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
+	int *total_seg, long order, char *name_buf, int *total_cpu, long *total_mem)
+{
+	basil_node_segment_t	*seg = NULL;
+	basil_node_processor_t	*proc = NULL;
+	basil_node_memory_t *mem = NULL;
+	basil_label_t *label = NULL;
+	basil_node_accelerator_t *accel = NULL;
+	basil_node_computeunit_t *cu = NULL;
+	int	aflag = READ_WRITE | ATR_DFLAG_CVTSLT;
+	long	totmem;
+	int	totcpus;
+	int	totaccel = 0;
+	int	first_seg = 0;
+	char	vname[VNODE_NAME_LEN];
+	char	*attr;
+	int	totseg;
+
+	/* Proceed only if we have valid pointers */
+	if (node == NULL) {
+		sprintf(log_buffer, "Bad pointer to node info");
+		log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_NODE,
```

Comment:
These log messages can't be documented , cause QA need to write and and verify the test case based on this EDD. And the current implementation is such that, these log messages will never get logged, hence can not be tested. I would suggest that we remove these log messages.